text/0000-wallet-replay-protection.md

```diff
@@ -0,0 +1,298 @@
+
+Let's say we have a transaction `T`
+```
+I1  -> anchor
+    -> O1
+    -> O2
+```
+
+We now use `O2` in a new transaction `T2` as an input to send money to Bob.
+
+```
+O2  -> O3 (change output)
+    -> Bob_output
+```
+
+The key idea here is to observe that `T2` can't be replayed by anyone else. Only we can create the input needed for the transaction to be replayed `O2`. The attacker could attempt to replay the previous transaction `T` to create the  `O2` input, but this would only be possible if they managed to replay transaction `T` which can't be replayed because it has an `anchor` output that has not been spent and hence can't be valid. This means that our transaction `T2` is protected against other parties replaying it. 
+
+If we again used `O3` as an input in a new transaction `T3` to send money to Charlie, the same protection would be put in place. The attacker would need to replay `T2` to recreate `O3` input, but we have already shown that `T2` can't be replayed.
+
+```
+O3  -> O4 (change output)
+    -> Charlie_output
+```
+
+This is the main idea used in this proposal to protect against replay attacks. This is why we name the outputs we created along side the `anchor` output `Protected` outputs. They provide protection for subsequent transactions we make if we use them as inputs to our transactions. Key thing to note here is that while we started with 2 protected outputs `O1` and `O2`, the outputs that we create in a transaction to which we add a protected input automatically become `Protected` outputs as well. This property is what allows us to continue creating new protected outputs without the need to create more than one `anchor` output.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Wallet rules for replay protection
+
+Most common transaction types are:
+1. A 1-2 `Regular` tx - 1 input, 2 outputs
+2. A 2-2 `PayJoin` tx - 2 inputs (one from each party), 2 outputs

```

Comment:
Payjoins are characterized by having inputs from both parties, not necessarily one each.
E.g. it's common that you need to combine multiple input to have a large enough amount to pay.
You might also want to spend additional unprotected outputs.